x/observer/migrations/v12/migrate.go

```diff
@@ -0,0 +1,58 @@
+package v12
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/zeta-chain/node/pkg/chains"
+	"github.com/zeta-chain/node/x/observer/types"
+)
+
+type observerKeeper interface {
+	GetChainParamsList(ctx sdk.Context) (val types.ChainParamsList, found bool)
+	SetChainParamsList(ctx sdk.Context, chainParams types.ChainParamsList)
+}
+
+// MigrateStore migrates the x/observer module state from the consensus version 11 to version 12.
+// The migration sets default gas price multipliers for all external chains.
+func MigrateStore(ctx sdk.Context, observerKeeper observerKeeper) error {
+	allChainParams, found := observerKeeper.GetChainParamsList(ctx)
+	if !found {
+		return errorsmod.Wrap(types.ErrChainParamsNotFound, "failed to get chain params")
+	}
+
+	// set new fields to the same value as 'confirmation_count'
+	for _, chainParams := range allChainParams.ChainParams {
+		chain, found := chains.GetChainFromChainID(chainParams.ChainId, []chains.Chain{})
+		if !found {
+			return errorsmod.Wrapf(types.ErrSupportedChains, "chain %d not found", chainParams.ChainId)
+		}
+
+		if chainParams != nil {
```

Comment:
Chain params cannot be nil here , I guess we can move this chech up if needed before using chainParams.ChainId

Reply:
fixed https://github.com/zeta-chain/node/pull/4485/commits/e387af05b0e2ba5e8b48852a8d741ae03aaef552